File: projectNew/old_src/TTA/TTAEventListener.java
package com.f14.TTA.listener;

import com.f14.TTA.TTAPlayer;
import com.f14.TTA.component.ability.EventAbility;

/**
 * TTA的事件监听器
 *
 * @author F14eagle
 */
public abstract class TTAEventListener extends TTAInterruptListener {
    /**
     * 事件卡
     */
    protected EventAbility ability;

    /**
     * 构造函数
     *
     * @param trigPlayer 触发玩家
     */
    public TTAEventListener(EventAbility ability, TTAPlayer trigPlayer) {
        super(trigPlayer);
        this.ability = ability;
    }

    /**
     * 取得监听事件的能力
     *
     * @return
     */
    public EventAbility getEventAbility() {
        return this.ability;
    }

}

File: projectNew/src/com/f14/TTA/listener/TTAEventListener.kt
package com.f14.TTA.listener

import com.f14.TTA.TTAGameMode
import com.f14.TTA.TTAPlayer
import com.f14.TTA.component.ability.EventAbility

/**
 * Created by 吹风奈奈 on 2017/7/20.
 */

abstract class TTAEventListener(gameMode: TTAGameMode, eventAbility: EventAbility, trigPlayer: TTAPlayer) : TTAInterruptListener(gameMode, trigPlayer) {
    /**
     * 事件卡
     */
    var eventAbility: EventAbility = eventAbility
        protected set
}
